src/QMCDrivers/MCPopulation.cpp

```diff
@@ -44,8 +44,44 @@ MCPopulation::MCPopulation(int num_ranks,
 
 MCPopulation::~MCPopulation() = default;
 
+void MCPopulation::copyHighMultiplicityWalkers()
+{
+  // we need to do this because spawnWalker changes walkers_ so we
+  // can't just iterate on that collection.
+  auto good_walkers = convertUPtrToRefVector(walkers_);
+  for (MCPWalker& good_walker : good_walkers)
+  {
+    int num_copies = static_cast<int>(good_walker.Multiplicity);
+    while (num_copies > 1)
+    {
+      auto walker_elements = spawnWalker();
+      // In the batched version walker IDs are set when walkers are born and are unique,
+      // ParentID is set to the walker that provided the initial configuration.
+      // If the amplified walker is was a transfer from another rank its copys get its ID as their parent.
+      // Not the transfered from ID that the received walker has.
+      // So the walker doesn't have to maintain state that it was a transfer and this code doesn't batch on it.
+      
+
+      // preserve the walkers id.  Ideally this wouldn't get overwritten but I don't think a custom assignment operator
+      // is worth it just for this.
+      auto walker_id         = walker_elements.walker.getWalkerID();
+      walker_elements.walker = good_walker;
+      // copy the copied from walkers id to parent id.
+      walker_elements.walker.setParentID(walker_elements.walker.getWalkerID());
+      // put the walkers actual id back.
+      walker_elements.walker.setWalkerID(walker_id);
+      // fix the multiplicity of the new walker
+      walker_elements.walker.Multiplicity = 1.0;
+      // keep good walker valid.
+      good_walker.Multiplicity -= 1.0;
+      num_copies--;
+    }
```

Comment:
This was the code that was in WalkerControl
```
for (size_t iw = 0; iw < good_walkers; iw++)
    {
      size_t num_copies = static_cast<int>(walkers[iw]->Multiplicity);
      while (num_copies > 1)
      {
        auto walker_elements = pop.spawnWalker();
        // save this walkers ID
        // \todo revisit Walker assignment operator after legacy drivers removed.
        // but in the modern scheme walker IDs are permanent after creation, what walker they
        // were copied from is in ParentID.
        long save_id                    = walker_elements.walker.getWalkerID();
        walker_elements.walker          = *walkers[iw];
        walker_elements.walker.setParentID(walker_elements.walker.getWalkerID());
        walker_elements.walker.setWalkerID(save_id);
        num_copies--;
      }
    }
```
The significant diferrence here is that we continue to make the multiplicities valid rather than just ignore them until the end when we set them all 1.0 regardless of what happened to them.  I'm of the opinion if you have a state variable its validity must be maintained through state changes not valid for some states and invalid in others.  That's asking for bugs later.